### ISSUE TITLE
Add support for macrocycle, mesocycle and microcycle

### ISSUE THREAD

**Issue by lediable** [wger-project/wger#310]
Hi!

It is possibile to add support for macro, meso and micro cycle training?

An example of structure is this:

Macro
   Meso 1
         Micro 1
         Micro 2
    Meso 2
         Micro 1
         Micro 2
         Micro 3

Macro is an annual plan, meso is phase of training with a duration of between 2-6 weeks and micro is instead only a week.

This is needed for sports periodization ( powerlifting, weightlifting, strongman, ecc ) or at least to create a fitness programs drafted on more than only one week.

**Comment by rolandgeider** [comment]
Hi there!

this is currently not supported, at least not directly. The "schedules" feature is similar to what you want, there you can add different plans and also select for how many weeks. The currently active plan will be selected according to this.

I know that it is not possible to represent every plan with this, specifically any planed weight increases or changes can't be modelled at all with the application. I had thought about doing something like this, but then I noticed it would become more complex than I anticipated. Before starting another try, we'd have to discuss what exactly would the macros, mesos and micros "do" and what the users would expect.

**Comment by lediable** [comment]
Hello!

Yes, the schedule does but indirectly.

We can let the user decide how to call and use these cycles.

We can give the possibility to create a training cycle and then put in the various sub (which will be the meso and micro) according to user needs.

Eg. The user chooses to create a training cycle. Him creates it. Then will be asked if him want to break it up into subcycle, the amount will be requested. 
Eventually it will be given the opportunity to create an additional subcycle (which in this case will be the weeks of training).

**Comment by dometto** [comment]
Think this also is superseded by https://github.com/wger-project/wger/issues/848

**Comment by rolandgeider** [comment]
Yes, you are right, as are all "workouts could be more flexible"
